tasty-query/shared/src/main/scala/tastyquery/reader/Loaders.scala

```diff
@@ -135,53 +139,109 @@ private[tastyquery] object Loaders {
+      def isNestedOrModuleClassName(name: String): Boolean =
+        def isNested =
+          val idx = name.lastIndexOf('$', name.length - 2)
+          idx >= 0 &&
+          !(idx + str.topLevelSuffix.length == name.length && name.endsWith(str.topLevelSuffix))
+        def isModule =
+          name.last == '$' && name.length > 1
+        isNested || isModule
+      end isNestedOrModuleClassName
+
+      if data.classes.isEmpty then
+        for
+          tData <- data.tastys
+          decodedName = NameTransformer.decode(tData.binaryName)
+          if !isNestedOrModuleClassName(decodedName)
+        do f(SimpleName(decodedName), Entry.TastyOnly(tData))
+      else
+        val tastyMap = data.tastys.map(t => t.binaryName -> t).toMap
+        for
+          cData <- data.classes
+          decodedName = NameTransformer.decode(cData.binaryName)
+          if !isNestedOrModuleClassName(decodedName)
+        do
+          val entry =
+            tastyMap.get(cData.binaryName).map(Entry.ClassAndTasty(cData, _)).getOrElse(Entry.ClassOnly(cData))
+          f(SimpleName(decodedName), entry)
+    end foldEntries
+
     def scanPackage(pkg: PackageSymbol)(using Context): Unit = {
       require(searched)
       packages.get(pkg) match {
-        case Some(data) =>
-          def isNestedOrModuleClassName(name: String): Boolean = {
-            def isNested = {
-              val idx = name.lastIndexOf('$', name.length - 2)
-              idx >= 0 &&
-              !(idx + str.topLevelSuffix.length == name.length && name.endsWith(str.topLevelSuffix))
-            }
-            def isModule =
-              name.last == '$' && name.length > 1
-            isNested || isModule
-          }
-
+        case Some(datas) =>
           packages -= pkg
-
           val localRoots = mutable.HashMap.empty[SimpleName, Entry]
-
-          if data.classes.isEmpty then
-            for
-              tData <- data.tastys
-              decodedName = NameTransformer.decode(tData.binaryName)
-              if !isNestedOrModuleClassName(decodedName)
-            do localRoots += (SimpleName(decodedName) -> Entry.TastyOnly(tData))
-          else
-            val tastyMap = data.tastys.map(t => t.binaryName -> t).toMap
-            for
-              cData <- data.classes
-              decodedName = NameTransformer.decode(cData.binaryName)
-              if !isNestedOrModuleClassName(decodedName)
-            do
-              val entry =
-                tastyMap.get(cData.binaryName).map(Entry.ClassAndTasty(cData, _)).getOrElse(Entry.ClassOnly(cData))
-              localRoots += (SimpleName(decodedName) -> entry)
-
+          for data <- datas do
+            foldEntries(data)((n, e) =>
+              localRoots.updateWith(n)({
+                case None        => Some(e) // first entry wins
+                case s @ Some(_) => s // do not replace existing root, i.e. follow classpath order
+              })
```

Comment:
Isn't that the same as
```suggestion
              localRoots.getOrElseUpdate(n, e)
```
?

Reply:
yes it seems so, thanks